src/lua/uri.lua

```diff
@@ -231,6 +231,16 @@ local function format(uri, write_password)
     uribuf.path = uri.path
     uribuf.query = uri.query
     uribuf.fragment = uri.fragment
+    -- IPv6 addresses need to be output encapsulated in brackets [].
+    -- Since gh-9556, we must set the `host_hint` to determine if `uri.host`
+    -- is an IPv6 address (refer to `src/lib/uri/uri.c:uri_format`). The IPv4
+    -- and Unix cases of `uri.host` do not require special formatting, so
+    -- `host_hint` is set to -1 as a catch-all value.
+    if uri.ipv6 ~= nil then
+        uribuf.host_hint = 2
+    else
+        uribuf.host_hint = -1
```

Comment:
It is zero by default (`uri_raw_parse` calls `memset`), not -1.

Since we anyway fill the `host_hint` value I would set 1 and 3 for `uri.ipv4` and `uri.unix`.

Reply:
Done